Source/WebCore/svg/graphics/filters/SVGFilter.cpp

```diff
@@ -308,6 +306,8 @@ RefPtr<FilterImage> SVGFilter::apply(FilterImage* sourceImage, FilterResults& re
 
             if (!sourceImage)
                 return nullptr;
+            else
+                effect->setOperatingColorSpace(sourceImage->colorSpace());
```

Comment:
But wait a second. Why we have to set the operating colorSpace of the `SourceGraphic` to the `colorSpace` of the `sourceImage`? The spec says
> The image is specified to be rendered in linear RGBA pixels.

And this is what we do when drawing the `sourceImage`. The ImageBuffer is created with `LinearRGB`. But the `SourceGraphic` can stay in `SRGB`. We will have to convert it only once.

In fact I think this change will introduce a pref regression when applying the SVGFilter. All the effects operate in `SRGB` unless the `effectElement.colorInterpolation()` says they should be in `LinearRGB`. So if you change the `SourceGraphic` to be `LinearRGB`, all the effects will have to convert the pixels of `SourceGraphic` to `SRGB` before operating on them.

Reply:
As you said the spec says:
> The image is specified to be rendered in linear RGBA pixels.

So the SourceGraphic (and SourceAlpha) should have operating color space depends on the `color-interpolation-filters` of the `Filter` element (which is by default linearRGB - spec: https://www.w3.org/TR/filter-effects-1/#propdef-color-interpolation-filters). Every filter effects can have its own `color-interpolation-filters` (but some of them ignore it, e.g: FEFlood (uses SRGB)).

So in case `<filter ...>` or `<filter color-interpolation-filters="linearRGB">` we should have:
* `sourceImage` in `linearRGB`
* SourceGraphic should operate in `linearRGB` so the output should be also in `linearRGB`

Example with `linearRGB`:
```
  <filter id="lightMe1">
    <feDiffuseLighting in="SourceGraphic" ...> or <feDiffuseLighting in="SourceGraphic" color-interpolation-filters="linearRGB" ...>...
```
* inputs of `feDiffuseLighting` should be transformed to linearRGB (SourceGraphic is in linearRGB so no-op)
* output of `feDiffuseLighting` should be also in linearRGB (also no-op)

Example with `sRGB`:
```
  <filter id="lightMe1">
    <feDiffuseLighting in="SourceGraphic" color-interpolation-filters="sRGB" ...>...
```
* inputs of `feDiffuseLighting` should be transformed to sRGB (SourceGraphic in this example is in linearRGB so it has to be transformed)
* output of `feDiffuseLighting` should be also in sRGB

Reply:
Anyway I will move this change (as @shallawa  suggested) to `buildFilterEffectsGraph()` and pass `DestinationColorSpace::LinearSRGB()` to `SourceGraphic::create()` if `filterElement.colorInterpolation() == ColorInterpolation::LinearRGB` (in other case will leave default value).

Reply:
It is indeed a complex topic. Would be great to get input from @shallawa / @heycam on the expected behavior.